network/src/peer_manager/mod.rs

```diff
@@ -762,7 +762,7 @@ where
                         );
                     }
                 } else {
-                    unreachable!(
+                    warn!(
                         "{} Received network event for unregistered protocol",
                         network_context
                     );
```

Comment:
Similarly with adding the offending types, it should help with debugging later :)